src/modules/nix_shell.rs

```diff
@@ -1,55 +1,66 @@
 use std::env;
 
-use super::{Context, Module, RootModuleConfig, SegmentConfig};
+use super::{Context, Module, RootModuleConfig};
 
 use crate::configs::nix_shell::NixShellConfig;
-
-// IN_NIX_SHELL should be "pure" or "impure" but lorri uses "1" for "impure"
-// https://github.com/target/lorri/issues/140
+use crate::formatter::StringFormatter;
 
 /// Creates a module showing if inside a nix-shell
 ///
 /// The module will use the `$IN_NIX_SHELL` and `$name` environment variable to
 /// determine if it's inside a nix-shell and the name of it.
 ///
 /// The following options are availables:
-///     - use_name   (bool)   // print the name of the nix-shell
 ///     - impure_msg (string) // change the impure msg
 ///     - pure_msg (string)   // change the pure msg
 ///
 /// Will display the following:
-///     - name (pure)    // use_name == true in a pure nix-shell
-///     - name (impure)  // use_name == true in an impure nix-shell
-///     - pure           // use_name == false in a pure nix-shell
-///     - impure         // use_name == false in an impure nix-shell
+///     - pure (name)    // $name == "name" in a pure nix-shell
+///     - impure (name)  // $name == "name" in an impure nix-shell
+///     - pure           // $name == "" in a pure nix-shell
+///     - impure         // $name == "" in an impure nix-shell
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let mut module = context.new_module("nix_shell");
     let config: NixShellConfig = NixShellConfig::try_load(module.config);
 
-    module.set_style(config.style);
-    module.create_segment("symbol", &config.symbol);
-
+    let shell_name = env::var("name").ok();
     let shell_type = env::var("IN_NIX_SHELL").ok()?;
-    let shell_type_segment: SegmentConfig = match shell_type.as_ref() {
-        "1" | "impure" => config.impure_msg,
+    let shell_type_format = match shell_type.as_ref() {
+        "impure" => config.impure_msg,
```

Comment:
Sorry, I'm not familiar with nix_shell. Does that mean `$IN_NIX_SHELL` will never be `1`?

Reply:
I'm not familiar with it either but it says in the docs that `$IN_NIX_SHELL` should only be `pure` or `impure`:
> IN_NIX_SHELL
>
>   Indicator that tells if the current environment was set up by nix-shell. Since Nix 2.0 the values are "pure" and "impure"

Reference: https://nixos.org/nix/manual/#sec-common-env